### PR TITLE
multi-term loss function capability: model loss arg can be dictionary…

### PR DESCRIPTION
**Pull request by bnb32**
… with multiple loss functions, and a list of weights.

### CODE REVIEW

**Review by grantbuster — CHANGES_REQUESTED** [review]

File: sup3r/utilities/loss_metrics.py [hunk 1]
```diff
@@ -7,6 +7,56 @@
 from tensorflow.keras.losses import MeanAbsoluteError, MeanSquaredError
 
 
+def _derivative(x, axis=1):
```
[the remark this answers is not in the document]
Comment (bnb32) [reply sup3r/utilities/loss_metrics.py:10]: This is using a central diff approx of the derivative though, not just difference of adjacent values.

File: sup3r/utilities/loss_metrics.py [hunk 2]
```diff
@@ -381,49 +351,7 @@ def __call__(self, x1, x2):
...
-            0D tensor with loss value
-        """
-        mae = self.MAE_LOSS(x1, x2)
-        ex_mae = self.EX_LOSS(x1, x2)
-
-        return (mae + 2 * self._weight * ex_mae) / 3
-
-
-class TemporalExtremesOnlyLoss(tf.keras.losses.Loss):
+class TemporalExtremesLoss(tf.keras.losses.Loss):
```
[the remark this answers is not in the document]
Comment (bnb32) [reply sup3r/utilities/loss_metrics.py:364]: Yeah, at least this is a simple json edit.

File: sup3r/utilities/loss_metrics.py [hunk 3]
```diff
@@ -305,17 +266,26 @@ def __call__(self, x1, x2, sigma=1.0):
...
+            'spatiotemporal data only. Received tensor(s) that are not 5D'
+        )
+        assert len(x1.shape) == 5 and len(x2.shape) == 5, msg
+
+        x1_div = tf.stack(
+            [self._compute_deriv(x1, fidx=i) for i in range(0, x1.shape[-1])]
+        )
+        x2_div = tf.stack(
+            [self._compute_deriv(x2, fidx=i) for i in range(0, x2.shape[-1])]
+        )
```
[the remark this answers is not in the document]
Comment (bnb32) [reply sup3r/utilities/loss_metrics.py]: Yeah, not sure why I did it this way haha

File: sup3r/utilities/loss_metrics.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (grantbuster) [reply sup3r/utilities/loss_metrics.py:10]: Are you convinced it's more important to do the central diff? I'm honestly not... Performance is a bit concerning for me but if you're okay with it then this is fine.

**Review by grantbuster — APPROVED** [review]
